modules/metalsmith/doc_assets.js

```diff
@@ -0,0 +1,168 @@
+                            error('nx_asset without file:content');
+                            return null;
+                        }
+
+                        // build the right url
+                        const folder = 'nx_assets';
+                        const ext = /(?:\.([^.]+))?$/.exec(doc.properties['file:content'].name)[1];
+                        const asset_type = doc.properties['doc_asset:nature'];
+                        const asset_file = folder + '/' + uid + '+' + asset_type + '.' + ext;
+                        const href = '/' + asset_file;
+
+                        $el.attr(attr, href);
+                        const html = $.html();
+                        file.contents = new Buffer(html);
+                        files[filename] = file;
+
+                        // if file already exist in fs
+                        if (fs.existsSync(asset_file)) {
+                            const content = fs.readFileSync(asset_file);
+                            const file_digest = checksum(content, doc.properties['file:content'].digestAlgorithm);
+
+                            // if digest matches ignore
+                            if (file_digest === doc.properties['file:content'].digest) {
+                                info('asset: %s already exists, ignoring', uid);
+                                return null;
+                            }
+                        }
+                        return doc.fetchRendition('docAsset')
+                            .then(function (res) {
+                                const writable = fs.createWriteStream(asset_file);
+                                res.body.pipe(writable);
+                                writable.on('finish', function () {
+                                    info('Downloaded asset: %s', uid);
+                                    writable.close();
+                                });
+                            });
+                    })
+                    .catch(function (err) {
+                        throw err;
+                    });
+
+                url_promises.push(p);
+            }
+
+            Promise.all(url_promises)
+                .then(() => {
+                    return resolve();
+                })
+                .catch((err) => {
+                    error(err);
+                    return resolve();
+                });
+        });
+
+    });
+};
+
+
+const doc_assets = function () {
+
+    return function (files, metalsmith, done) {
+
+        const nuxeo = new Nuxeo({
+            baseURL: process.env.NX_ASSETS_URL || 'http://localhost:8080/nuxeo',
+            auth   : {
+                method  : 'basic',
+                username: process.env.NX_ASSETS_USER || 'Administrator',
+                password: process.env.NX_ASSETS_PWD || 'Administrator'
+            },
+        });
+
+        const file_promises = [];
+
+        for (const filename in files) {
```

Comment:
My preference would be a functional style and move the `selector` definition outside of the loop.
```js
const selector = Object.keys(map).join();
Object.keys(files).forEach((filename) => {
  if (is_html(filename)) {
    file_promises.push(check_file(files, filename, selector, nuxeo));
  }
})
```
It's not wrong, just style preference.